include/openthread/joiner.h

```diff
@@ -111,11 +111,8 @@ otError otJoinerStart(otInstance *     aInstance,
  *
  * @param[in]  aInstance  A pointer to an OpenThread instance.
  *
- * @retval OT_ERROR_NONE              Successfully disabled the Joiner role.
- * @retval OT_ERROR_DISABLED_FEATURE  The Joiner feature is not enabled in this build.
- *
  */
-otError otJoinerStop(otInstance *aInstance);
+void otJoinerStop(otInstance *aInstance);
```

Comment:
Shall we keep the return type untouched so that `SuccessOrExit(otJoinerStop(...));` pattern won't cause a compile error?

Reply:
Good question! @abtink @jwhui Thoughts?

Reply:
Here just propose another topic might be a little broader.
Currently it seems there are different strategies regarding the APIs specific to the configurable options, for example
otJoinerXXX() APIs would have empty body when the option is not enabled, while 
all otServerXXX() APIs are just wrapped within the option macro 
https://github.com/openthread/openthread/blob/master/src/core/api/server_api.cpp#L36

Are there any specific rules to follow? @jwhui @bukepo